### PR TITLE
Dev skip minification

### PR DESCRIPTION
**Pull request by bramburn**
Added an option to skip minification by adding <!— skip.minification —>

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Looks good to me. Please add a test, then I'll merge. 👍

**Comment by bramburn** [comment]
@GrahamCampbell i've added a test, I've never created tests before. Hopefully its up to your satisfaction.

I normally just code for myself and I just test while I code. This is all new to me. Hope this helps.

**Comment by bramburn** [comment]
@GrahamCampbell let me know if you want additional tests

File: src/Minifiers/BladeMinifier.php [hunk 1]
```diff
@@ -94,6 +97,8 @@ protected function containsBadHtml($value)
     {
         return preg_match('/<(code|pre|textarea)/', $value) ||
             preg_match('/<script[^\??>]*>[^<\/script>]/', $value) ||
+            preg_match('/skip\.minification/', $value) ||
+            preg_match('/Schema::create/', $value) ||
```
Comment (GrahamCampbell) [inline src/Minifiers/BladeMinifier.php]: Please don't add this. This is very specific, and there's no guarantee that that that alias name is even used.

Comment (bramburn) [inline src/Minifiers/BladeMinifier.php]: you mean Schema::create? or skip.minification?

Comment (bramburn) [inline src/Minifiers/BladeMinifier.php]: i have removed the Schema::create one.

**Comment by m1guelpf** [comment]
Updates?

**Comment by bramburn** [comment]
@m1guelpf Graham doesn't want to approve this as the method is not for everyone. You can fork my branch if you want.
https://github.com/bramburn/Laravel-HTMLMin
It works for general HTML use.
Graham did mention it could be faster to just generate the HTML min on the fly.

File: tests/Functional/stubs/skipmarkdown.blade.php [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+hello world!
+
+
+* one 
+* two 
+* three
+
+skip minification
+<!-- skip.minification -->
```
Comment (GrahamCampbell) [inline tests/Functional/stubs/skipmarkdown.blade.php]: missing new line

File: tests/Functional/SkipMinificationTest.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+/**
+ * This is the blade enabled test class.
+ *
+ * @author Graham Campbell <graham@alt-three.com>
+ */
+class SkipMinificationTest extends AbstractFunctionalTestCase
+{
+    /**
+     * Setup the application environment.
+     * Ensure that blade is on.
```
Comment (GrahamCampbell) [inline tests/Functional/SkipMinificationTest.php]: Please add a newline before this line.

File: tests/Functional/stubs/test.blade.php [hunk 4]
```diff
@@ -19,4 +19,4 @@
         <p>A simple HTML minifier for Laravel 5</p>
     </div>
 </body>
-</html>
+</html>
```
Comment (GrahamCampbell) [inline tests/Functional/stubs/test.blade.php]: missing newline

File: src/Minifiers/BladeMinifier.php [hunk 5]
```diff
@@ -94,6 +97,7 @@ protected function containsBadHtml($value)
     {
         return preg_match('/<(code|pre|textarea)/', $value) ||
             preg_match('/<script[^\??>]*>[^<\/script>]/', $value) ||
+            preg_match('/skip\.minification/', $value) ||
```
Comment (GrahamCampbell) [inline src/Minifiers/BladeMinifier.php]: This matches things that aren't comments too?

Comment (bramburn) [inline src/Minifiers/BladeMinifier.php]: i think this needs to be comment only

File: tests/Functional/SkipMinificationTest.php (hunk 3 above)
Comment (bramburn) [inline tests/Functional/SkipMinificationTest.php]: done

File: tests/Functional/stubs/test.blade.php (hunk 4 above)
Comment (bramburn) [inline tests/Functional/stubs/test.blade.php]: done

File: src/Minifiers/BladeMinifier.php (hunk 5 above)
Comment (bramburn) [inline src/Minifiers/BladeMinifier.php]: done

File: tests/Functional/SkipMinificationTest.php [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+ */
+
+namespace GrahamCampbell\Tests\HTMLMin\Functional;
+
+use Illuminate\Contracts\Foundation\Application;
+
+/**
+ * This is the blade enabled test class.
+ *
+ * @author Graham Campbell <graham@alt-three.com>
```
Comment (GrahamCampbell) [inline tests/Functional/SkipMinificationTest.php]: You can replace my name with your name for this file. :)

File: tests/Functional/stubs/skipmarkdown.blade.php [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+hello world!
+
+
+* one 
+* two 
+* three
+
+skip minification
+
+<!-- skip.minification -->
```
Comment (GrahamCampbell) [inline tests/Functional/stubs/skipmarkdown.blade.php]: Missing newline.

File: tests/Functional/stubs/skiptest.blade.php [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+<body>
+    <div class="container">
+        <h1>Laravel HTMLMin Test Page</h1>
+
+        <p>A simple HTML minifier for Laravel 5</p>
+    </div>
+</body>
+
+</html>
+<!-- skip.minification -->
```
Comment (GrahamCampbell) [inline tests/Functional/stubs/skiptest.blade.php]: Missing newline.

File: tests/Functional/stubs/test.blade.php [hunk 9]
```diff
@@ -19,4 +19,5 @@
         <p>A simple HTML minifier for Laravel 5</p>
     </div>
 </body>
-</html>
+
+</html>
```
Comment (GrahamCampbell) [inline tests/Functional/stubs/test.blade.php]: Missing newline.

**Comment by GrahamCampbell** [comment]
Nearly good to merge. :)

**Comment by GrahamCampbell** [comment]
Actually, will just merge as is, those other things are so minor. Thanks for the PR. :)

**Pull request merged**